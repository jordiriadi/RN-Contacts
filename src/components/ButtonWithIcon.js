import React from 'react';
import {
  StyleSheet, Text, TouchableOpacity, Dimensions, View, ViewPropTypes,
} from 'react-native';
import PropTypes from 'prop-types';
import IonIcons from 'react-native-vector-icons/Ionicons';
import Styles from '../utils/Styles';

const ButtonWithIcon = (props) => {
  const { disabled, iconName, iconSize = 25, buttonText } = props;
  return (
    <TouchableOpacity
      id="btn-container"
      disabled={disabled}
      onPress={props.onPress}
      style={{
        ...styles.button,
        ...props.style,
        backgroundColor: disabled
          ? Styles.DisabledBtnBackgroundColor
          : styles.button.backgroundColor,
      }}
    >
      <View style={styles.buttonContainer}>
        <IonIcons
          style={{ color: Styles.SecondaryColor }}
          name={iconName}
          size={iconSize}
        />
        <Text style={styles.buttonText}>{buttonText}</Text>
      </View>
    </TouchableOpacity>
  );
};

const { height } = Dimensions.get('window');

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Styles.PrimaryButtonColor,
    height: height / 15,
    padding: Styles.DefaultPadding,
  },
  buttonText: {
    ...Styles.DefaultTitleText,
    color: 'white',
  },
  buttonContainer: {
    alignItems: 'center'
  }
});

ButtonWithIcon.propTypes = {
  style: ViewPropTypes.style,
  disabled: PropTypes.bool,
  onPress: PropTypes.func,
  iconName: PropTypes.string,
  iconSize: PropTypes.number,
  buttonText: PropTypes.string,
};

export default ButtonWithIcon;
