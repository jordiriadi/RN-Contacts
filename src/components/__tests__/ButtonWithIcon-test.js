import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';

import ButtonWithIcon from '../ButtonWithIcon';

describe('ButtonWithIcon Component', () => {  
  it('renders correctly', () => {
  
    const wrapper = shallow(<ButtonWithIcon/>);

    expect(wrapper).toMatchSnapshot();
  });

})