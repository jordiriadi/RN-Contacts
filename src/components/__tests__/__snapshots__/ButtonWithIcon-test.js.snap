// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithIcon Component renders correctly 1`] = `ShallowWrapper {}`;
